config/archive_test.go

```diff
@@ -0,0 +1,120 @@
+package config
+
+import (
+	"io/ioutil"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/contiv/volplugin/watch"
+	. "gopkg.in/check.v1"
+)
+
+const (
+	testPolicyName = "foobar"
+)
+
+func revisionCount(s *configSuite) int {
+	revisions, _ := s.tlc.ListPolicyRevisions(testPolicyName)
+	return len(revisions)
+}
+
+func createTestPolicy(s *configSuite, c *C) {
+	err := s.tlc.CreatePolicyRevision(testPolicyName, "{}")
+	c.Assert(err, IsNil)
+}
+
+func (s *configSuite) TestCreatePolicyRevision(c *C) {
+	c.Assert(revisionCount(s), Equals, 0)
+	createTestPolicy(s, c)
+	c.Assert(revisionCount(s), Equals, 1)
+}
+
+func (s *configSuite) TestListPolicyRevisions(c *C) {
+	c.Assert(revisionCount(s), Equals, 0)
+
+	// edge case: the second may roll over between saving the current timestamp
+	//            and the HTTP request to upload the policy being handled
+	t := int(time.Now().Unix())
+	t2 := t + 1
+
+	createTestPolicy(s, c)
+	revisions, err := s.tlc.ListPolicyRevisions(testPolicyName)
+	c.Assert(err, IsNil)
+	c.Assert(len(revisions), Equals, 1)
+
+	revInt, err := strconv.Atoi(revisions[0])
+	c.Assert(err, IsNil)
+	c.Assert(revInt == t || revInt == t2, Equals, true)
+}
+
+func (s *configSuite) TestGetPolicyRevision(c *C) {
+	policyText := `{"foo":"bar"}`
+
+	c.Assert(revisionCount(s), Equals, 0)
+
+	err := s.tlc.CreatePolicyRevision(testPolicyName, policyText)
+	c.Assert(err, IsNil)
+
+	revisions, err := s.tlc.ListPolicyRevisions(testPolicyName)
+	c.Assert(err, IsNil)
+	c.Assert(len(revisions), Equals, 1)
+
+	revision := revisions[0]
+
+	resp, err := s.tlc.GetPolicyRevision(testPolicyName, revision)
+	c.Assert(err, IsNil)
+	c.Assert(resp, Equals, policyText)
+}
+
+// this test ensures that a new policy being uploaded results in the correct
+// data (policy name + timestamp, tab-separated) being written to stdout
+func (s *configSuite) TestWatchForPolicyChanges(c *C) {
+	s.tlc.WatchForPolicyChanges(nil)
+
+	original := os.Stdout
+	r, w, err := os.Pipe()
+	c.Assert(err, IsNil)
+
+	// edge case: the second may roll over between saving the current timestamp
+	//            and the HTTP request to upload the policy being handled
+	t := int(time.Now().Unix())
+	t2 := t + 1
+
+	// run the stdout manipulation code in a func so we can guarantee a defer to
+	// reset everything runs immediately afterwards even in a panic() situation
+	func() {
```

Comment:
this `func` is superfluous... not quite sure what you're trying to accomplish here. Do you just need a new scope?